tests/test_spark_utils.py

```diff
@@ -18,3 +21,8 @@ def test_try_read_dataframe_valid(tmp_path):
     df = try_read_dataframe(filename)
     assert isinstance(df, DataFrame)
     assert df.collect()[0].a == 1
+
+
+def test_get_type_from_union():
+    assert get_type_from_union(Optional[str]) is str
+    assert get_type_from_union(Optional[DayType]) is str
```

Comment:
So Optional[X] is the same as Union[X,None]?

Reply:
yes